### DISCUSSION TITLE
NX + TRPC + Client

### DISCUSSION THREAD

**Discussion by emilianomon** [trpc/trpc#2257] [Q&A]
Hi guys! Soon to be fan here. 🙌

I'd like to check with you guys if you already have a working example of two apps (client and server) coexisting and successfully sharing the AppRouter type using NX.

In my case, I have a RN app importing the RootRouter (AppRouter alias) from an Express app. Both living in the apps folder. But as you can see in the images bellow, the server defines and exports the type as it should, but right on the import, the client does not seem to be able to properly infer the type.

![image](https://user-images.githubusercontent.com/84454335/179645783-c4d04bd6-cf42-41a8-9624-ff8a837d36d6.jpeg)

![image](https://user-images.githubusercontent.com/84454335/179645814-1360538f-63b1-4938-a5af-2a9edbd18790.jpeg)

Anyone has any idea of what might be happening? Maybe a configuration is missing?

Thanks in advance.

**Comment by emilianomon** [comment]
I've managed to find the problem, and it was by trial and error. Really hard problem to point out since it seems a Typescript issue/limitation.

For some reason, if you create a wrapper between tRPC routing and the application, the type gets lost but only when importing from another application. I know it seems weird but In my case I had `export const createRouter = () => trpc.router();`. The problem went away just by using `trpc.router()` directly.

**Reply by eduardoffelipe** [reply]
zod fuck  yout life 👯

**Comment by ixahmedxi** [comment]
I also have an nx trpc monorepo and what I did was to completely develop the trpc API in a separate library where it exports the appRouter and the type of the AppRouter. This is used on the server to then make a nextjs api handler, in your case it would be to connect the appRouter to express and then you can import the AppRouter as well in your client.
